core/trino-main/src/main/java/io/trino/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -111,9 +111,12 @@ public double getUtilization()
     }
 
     @Override
-    public boolean isOverutilized()
+    public OutputBufferStatus getStatus()
     {
-        return (getUtilization() > 0.5) && stateMachine.getState().canAddPages();
+        // do not grab lock to acquire outputBuffers to avoid delaying TaskStatus response
+        return OutputBufferStatus.builder(outputBuffers.getVersion())
+                .setOverutilized(memoryManager.getUtilization() >= 0.5 || !stateMachine.getState().canAddPages())
```

Comment:
this changes logic - new version looks ok. Was there a bug? Looks like fix itself should be a separate commit.

Reply:
Good catch. Let me revert it to what it were. Changing the logic wasn't an intention I had.